pkgs/top-level/all-packages.nix

```
@@ -30478,7 +30478,10 @@ with pkgs;
 
   ninjas2 = callPackage ../applications/audio/ninjas2 {};
 
-  nncp = callPackage ../tools/misc/nncp { };
+  nncp = (
+    if stdenv.isDarwin
+    then darwin.apple_sdk_11_0.callPackage
+    else callPackage) ../tools/misc/nncp { };
```

This works for all systems, but affects darwin only.
```suggestion
  nncp = darwin.apple_sdk_11_0.callPackage ../tools/misc/nncp { };
```
It's not the best solution ever 🤷 but it's necessary for some packages. (see others like `_7zz` or `mlterm` in this file)

---

A look at `doc/stdenv/platform-notes.chapter.md` and the definition of even go itself in `all-packages.nix` seems to confirm this. So if this fixes the darwin build and if the linux build is still fine, go for it.

---

Sorry to jump in, I was looking at a different package with same error and this didn't help
```patch
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 51a52a44915..9d74ffcd8dd 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -16353,7 +16353,7 @@ with pkgs;
 
   bazel-remote = callPackage ../development/tools/build-managers/bazel/bazel-remote { };
 
-  bazel-watcher = callPackage ../development/tools/bazel-watcher {
+  bazel-watcher = (if stdenv.isDarwin then darwin.apple_sdk_11_0.callPackage else callPackage) ../development/tools/bazel-watcher {
     go = go_1_18;
   };
```
on x86_64-darwin it still gets
```
In file included from /nix/store/gvz6j31hm9hg44xiwb503rbv2gc1x0nw-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/Security.h:87:
/nix/store/gvz6j31hm9hg44xiwb503rbv2gc1x0nw-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecCode.h:35:10: fatal error: 'xpc/xpc.h' file not found
#include <xpc/xpc.h>
         ^~~~~~~~~~~
1 error generated.
compilepkg: error running subcommand external/go_sdk/pkg/tool/darwin_amd64/cgo: exit status 2
Target //ibazel:ibazel failed to build
```
any hints?

---

This shouldn't have been merged like this. Going to fix it up right now.

---

No, I'm not using darwin and I don't want my packages fed through darwin functions.

---

Then the apple 11 sdk must be removed instead.